store/tikv/prewrite.go

```diff
@@ -140,6 +151,27 @@ func (action actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoff
 					c.run(c, nil)
 				}
 			}
+
+			if c.isOnePC() {
+				if prewriteResp.OnePcCommitTs == 0 {
+					logutil.Logger(bo.ctx).Warn("1pc failed and fallbacks to normal commit procedure",
+						zap.Uint64("startTS", c.startTS))
+					tikvOnePCTxnCounterFallback.Inc()
+					c.setOnePC(false)
+				} else {
+					c.mu.Lock()
```

Comment:
No need to lock it because either it runs in the same goroutine or it's synced by channel:
> A send on a channel happens before the corresponding receive from that channel completes.